activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -385,6 +385,28 @@ def column_name_for_operation(operation, node) # :nodoc:
         visitor.accept(node, collector).value
       end
 
+      def table_name_options(config = ActiveRecord::Base)
+        {
+          table_name_prefix: config.table_name_prefix,
+          table_name_suffix: config.table_name_suffix
+        }
+      end
+
+      # Finds the correct table name given an Active Record object.
+      # Uses the Active Record object's own table_name, or pre/suffix from the
+      # options passed in.
+      def proper_table_name(name, options = table_name_options)
+        if name.respond_to? :table_name
+          name.table_name
+        else
+          "#{options[:table_name_prefix]}#{name}#{options[:table_name_suffix]}"
+        end
+      end
+
+      def remove_prefix_and_suffix(table)
```

Comment:
I assume you mean class? That was my plan originally, but since I took it out of migration where it was a instance method I kept it like that.

The only downside to a class method would be all of the calls to it would change from AR.connection to AR.connection.class

I'll wait for feedback on this.